conans/client/tools/oss.py

```diff
@@ -202,69 +202,71 @@ class _OSVERSIONINFOEXW(ctypes.Structure):
     def get_debian_version_name(version):
         if not version:
             return None
-        elif version.major() == "8.Y.Z":
+        elif version.major == 8:
             return "jessie"
-        elif version.major() == "7.Y.Z":
+        elif version.major == 7:
             return "wheezy"
-        elif version.major() == "6.Y.Z":
+        elif version.major == 6:
             return "squeeze"
-        elif version.major() == "5.Y.Z":
+        elif version.major == 5:
             return "lenny"
-        elif version.major() == "4.Y.Z":
+        elif version.major == 4:
             return "etch"
-        elif version.minor() == "3.1.Z":
+        elif (version.major, version.minor) == (3, 1):
             return "sarge"
-        elif version.minor() == "3.0.Z":
+        elif (version.major, version.minor) == (3, 0):
             return "woody"
 
     @staticmethod
     def get_win_version_name(version):
         if not version:
             return None
-        elif version.major() == "5.Y.Z":
+        elif version.major == 5:
             return "Windows XP"
-        elif version.minor() == "6.0.Z":
+        elif version.major == 6 and version.minor == 0:
             return "Windows Vista"
-        elif version.minor() == "6.1.Z":
+        elif version.major == 6 and version.minor == 1:
             return "Windows 7"
-        elif version.minor() == "6.2.Z":
+        elif version.major == 6 and version.minor == 2:
             return "Windows 8"
-        elif version.minor() == "6.3.Z":
+        elif version.major == 6 and version.minor == 3:
             return "Windows 8.1"
-        elif version.minor() == "10.0.Z":
+        elif version.major == 10:
             return "Windows 10"
+        elif version.major == 11:
```

Comment:
Looking deeper into this, I'm not sure about this anymore.

Reply:
Not a big deal though. We will have a look to this legacy code here, and probably hide or even remove some of it. I have been having a look at conan-center-index, and it is not very clear what is the usage of the Windows version detection (if the Windows version is relevant, it should probably go in a subsetting, but not automatically detected by a recipe. This is the criteria we are following for 2.0)